pygeoapi/provider/filesystem.py

```diff
@@ -275,8 +285,22 @@ def _describe_file(filepath):
                     [d.bounds[0], d.bounds[1]]
                 ]]
             }
+        
+        model = {}
         for k, v in d.schema['properties'].items():
-            content['properties'][k] = v
+            model[k] = v
+        if model.items():
+            content['properties']['model'] = model
+
+        from pygeoapi.util import yaml_load
```

Comment:
In terms of a metadata sidecar file, do we have anything more formal? An OARec record model would be perfect here (once it is ready that is).

Reply:
+1 for more formal, I used the stac model for now.

Reply:
@tomkralidis would you suggest to use pygeometa or ogc-api-records format? Or even dcat (as json-ld). Any of those would need a transformation for use in stac, but is more future proof if at some point we want to use the filesystem provider as a source for ogc-api-records

Reply:
@pvgenuchten ideally we would allow a user to specify what the side car files are in config (i.e. xml, yaml), as well as their representation so then pygeoapi can handle accordingly.  Having said this, perhaps pygeometa [MCF files](https://github.com/geopython/pygeometa/blob/master/sample.yml) will do the trick here (of course MCF file properties may need some tweaking over in pygeometa).

Reply:
I prefer a limited set of formats, no on the fly metadata conversion from various formats, there's other tools for that. A scheduled process can run over the files and prepare mcf files for each.

Actually you can already add mcf with this PR, it will however not be transformed to the model required by STAC. 

I propose to merge this and continue the work in next PR's, what do you think?

Reply:
> ideally we would allow a user to specify what the side car files are in config (i.e. xml, yaml), as well as their representation so then pygeoapi can handle accordingly. 

We could (similar to how spatial file types are detected) add a configuration specifying what type of file extensions are made available as metadata files, potentially defining a transformer/formatter function per file type...

Reply:
What about the following:
- enable pygeoapi to look for a filename with `.yml` extension, which is an MCF file
- add capability to pygeometa to export an MCF file as a STAC item, which would then provide the STAC item representation back to pygeoapi, like:
```python
        from pygeometa.core import read_mcf, MCFReadError

        pth = os.path.splitext(filepath)[0]
        try:
            md = read_mcf(pth)
            content = md.to_stac_item()  # not implemented yet
        except MCFReadError as err:
            LOGGER.warning(err)
```

Reply:
Nice! 
But is it a responsibility of the provider? It means the provider can only be used by STAC. I hope we can also re-use this provider for OAPI-Records or even OAPI-Features at some point...

Reply:
Good point. For OARec and STAC API, wouldn't this imply a search index, from which we would have pygeoapi scrape the directory tree to build up an index (which would happen outside of this provider)?

Reply:
Correct, my impression is that this provider caches the tree structure in memory, which seem ok for 100-ish files, but will be challenging if it is 1000's. Besides that, it now retrieves some metadata, such as dataset bounds at opening the record, which we would need prior to that for the whole tree if we want to enable spatial queries on the index, i already noticed this ogrinfo into files is very time & resource consuming, so for sure something you want to cache through restarts.

We then come to a discussion: is filesystem really a provider? Maybe it is even more a process (systemSpider) which prepares data for pygeoapi in an elastic or alternative index and both STAC and OAPI-Records engage with the index only.

Reply:
Maybe we can use OAPI-Process to trigger the rebuild of the index 😄